src/rpc/budget.cpp

```diff
@@ -48,73 +48,85 @@ void budgetToJSON(CBudgetProposal* pbudgetProposal, UniValue& bObj)
     bObj.push_back(Pair("fValid", pbudgetProposal->fValid));
 }
 
-UniValue preparebudget(const UniValue& params, bool fHelp)
+void checkBudgetInputs(const UniValue& params, std::string &strProposalName, std::string &strURL,
+                       int &nPaymentCount, int &nBlockStart, CBitcoinAddress &address, CAmount &nAmount)
 {
     int nBlockMin = 0;
     CBlockIndex* pindexPrev = chainActive.Tip();
```

Comment:
I think calling `chainActive.Tip()` should be guarded with a LOCK.

Reply:
It looks like there is a lot of use where it's not actually guarded; so this definitely calls out a much bigger potential issue then just this area of code; especially places where it's called multiple times and may not handle if it changes between the calls (a good thing to investigate).

In this specific case, however, the tip is only used to figure out if the block being requested for the budget payment is in the future, and an actual budget block.  So if Tip isn't dead on, I don't think it matters.